tests/Microsoft.Identity.Test.Unit.net45/CacheTests/UnifiedSchemaValidationTests.cs

```diff
@@ -0,0 +1,842 @@
+            var expectedJsonObject = new JObject
+            {
+                ["local_account_id"] = "00000000-0000-0000-40c0-3bac188d01d1",
+                ["home_account_id"] = "00000000-0000-0000-40c0-3bac188d01d1.9188040d-6c67-4c5b-b112-36a304b66dad",
+                ["username"] = "msalsdktest@outlook.com",
+                ["environment"] = "login.microsoftonline.com",
+                ["realm"] = "9188040d-6c67-4c5b-b112-36a304b66dad",
+                ["authority_type"] = "MSSTS",
+                ["name"] = "Test Username",
+                ["client_info"] = "eyJ2ZXIiOiIxLjAiLCJzdWIiOiJBQUFBQUFBQUFBQUFBQUFBQUFBQUFNTmVBRnBTTGdsSGlPVHI5SVpISkVBIiwibmFtZSI6Ik9sZ2EgRGFsdG9tIiwicHJlZmVycmVkX3VzZXJuYW1lIjoibXNhbHNka3Rlc3RAb3V0bG9vay5jb20iLCJvaWQiOiIwMDAwMDAwMC0wMDAwLTAwMDAtNDBjMC0zYmFjMTg4ZDAxZDEiLCJ0aWQiOiI5MTg4MDQwZC02YzY3LTRjNWItYjExMi0zNmEzMDRiNjZkYWQiLCJob21lX29pZCI6IjAwMDAwMDAwLTAwMDAtMDAwMC00MGMwLTNiYWMxODhkMDFkMSIsInVpZCI6IjAwMDAwMDAwLTAwMDAtMDAwMC00MGMwLTNiYWMxODhkMDFkMSIsInV0aWQiOiI5MTg4MDQwZC02YzY3LTRjNWItYjExMi0zNmEzMDRiNjZkYWQifQ"
+            };
+
+            AssertAreJsonStringsEquivalent(expectedJsonObject.ToString(), credential.ToJsonString());
+
+            MsalAccountCacheKey key = credential.GetKey();
+
+            string expectedServiceKey = "9188040d-6c67-4c5b-b112-36a304b66dad";
+            Assert.AreEqual(expectedServiceKey, key.iOSService);
+
+            string expectedAccountKey = "00000000-0000-0000-40c0-3bac188d01d1.9188040d-6c67-4c5b-b112-36a304b66dad-login.microsoftonline.com";
+            Assert.AreEqual(expectedAccountKey, key.iOSAccount);
+
+            string expectedGenericKey = "msalsdktest@outlook.com";
+            Assert.AreEqual(expectedGenericKey, key.iOSGeneric);
+
+            Assert.AreEqual(1003, key.iOSType);
+        }
+
+        [TestMethod]
+        public void TestSchemaComplianceForAccessToken_WhenMSSTSResponse_WithB2CAccount()
+        {
+            var credential = new MsalAccessTokenCacheItem(MsalEnvironment, B2CClientId, CreateB2CTestTokenResponse(), B2CTenantId);
+            DateTime currentDate = DateTime.UtcNow;
+            string expiresOn = CoreHelpers.DateTimeToUnixTimestamp(currentDate.AddSeconds(3600));
+            string extExpiresOn = CoreHelpers.DateTimeToUnixTimestamp(currentDate.AddSeconds(262800));
+            string cachedAt = CoreHelpers.DateTimeToUnixTimestamp(currentDate);
+
+            // 1. Verify payload
+            var expectedJsonObject = new JObject
+            {
+                ["secret"] = "<removed_at>",
+                ["target"] = "https://iosmsalb2c.onmicrosoft.com/webapitest/user.read",
+                ["extended_expires_on"] = extExpiresOn,
+                ["ext_expires_on"] = extExpiresOn,
+                ["credential_type"] = "AccessToken",
+                ["environment"] = "login.microsoftonline.com",
```

Comment:
this is confusing, because it's a b2c account...if i run the desktop test w/b2c, i get: `msidlabb2c.b2clogin.com` as the environment...and login.microsoftonline.com is being deprecated, so we should use `something.b2clogin.com` and what about custom domains?